packages/core/src/helpers/map-refs.ts

```diff
@@ -75,9 +75,30 @@ export const mapRefs = (
   for (const key of Object.keys(
     component.hooks,
   ) as (keyof typeof component.hooks)[]) {
-    const hookCode = component.hooks[key]?.code;
-    if (hookCode) {
-      component.hooks[key]!.code = replaceRefsInString(hookCode, refs, mapper);
+    if (Array.isArray(component.hooks[key])) {
+      (component.hooks[key] as extendedHook[]).forEach((hook) => {
+        if (hook.code) {
+          hook.code = replaceRefsInString(hook.code, refs, mapper);
+        }
+      });
+    } else {
+      const hookCode = (component.hooks[key] as extendedHook)?.code;
+      if (hookCode) {
+        (component.hooks[key] as extendedHook)!.code = replaceRefsInString(
+          hookCode,
+          refs,
+          mapper,
+        );
+      }
```

Comment:
We can avoid the need for the type-casts here if we create a `const hooks = component.hooks[key];` before the if-else logic

```suggestion
    const hooks = component.hooks[key];
    if (Array.isArray(hooks)) {
      hooks.forEach((hook) => {
        if (hook.code) {
          hook.code = replaceRefsInString(hook.code, refs, mapper);
        }
      });
    } else {
      const hookCode = hooks?.code;
      if (hookCode) {
        hooks.code = replaceRefsInString(hookCode, refs, mapper);
      }
    }
```